op-scim-bridge/chart/op-scim-bridge/templates/deployer-cleanup.yaml

```diff
@@ -1,15 +1,22 @@
 apiVersion: batch/v1beta1
 kind: CronJob
 metadata:
-  name: deployer-cleanup # deletes stale deployers to improve upgradeability of the SCIM bridge
+  name: "{{ .Values.name }}-cleanup" # deletes stale deployers to improve upgradeability of the SCIM bridge
+  labels: &CronLabels
+    app.kubernetes.io/name: {{ .Values.name }}
+    app.kubernetes.io/component: cleanup
 spec:
   schedule: "*/1 * * * *" # once a minute
+  successfulJobsHistoryLimit: 0
+  failedJobsHistoryLimit: 0
   jobTemplate:
     spec:
       template:
+        metadata:
+          labels: *CronLabels
         spec:
           containers:
           - name: kubectl-runner
             image: bitnami/kubectl
-            command: ["bash", "-c", "if [[ `kubectl get pods -n {{ .Values.namespace }} | grep -i deployer | awk '$2 ~ 1/1' | awk '{print $1}'` ]]; then kubectl delete jobs -n {{ .Values.namespace }} $(kubectl get jobs -n {{ .Values.namespace }} | grep -i deployer | awk '$2 ~ 1/1' | awk '{print $1}'); fi"]
+            command: ["bash", "-c", "if [[ `kubectl get pods -n {{ .Values.namespace }} | awk '$1 ~ /{{ .Values.name }}-deployer/ && $2 ~ /0.*1/ { print $1 }'` ]]; then kubectl delete jobs -n {{ .Values.namespace }} $(kubectl get jobs -n {{ .Values.namespace }} | awk '$1 ~ /{{ .Values.name }}-deployer/ && $2 ~ /0.*1/ {print $1}' 2>/dev/null) 2>/dev/null; fi; exit 0"]
```

Comment:
The diff here is a bit hard to review; it's just commands piped together. Is there a cleaner way to go about doing this? EG: where the changes will be presented in a nice structured file rather than a single line.

Reply:
I could break it up into multiple lines. Would that work?

Reply:
That would probably be a bit clearer. However at the end of the day, I'll leave the format up to you. Whatever you think is the most clear and best outcome.

I just wanted to note that this is somewhat hard to parse as a viewer, and that if we are doing such a complex task in a `command` block, that smells to me as if we are not doing something right. However I don't know what the 'right' way to do this is either. 

I'll do a little looking around to see if there is a better way to do this.

Reply:
I looked high and low to see if there was a better solution. Most solutions seem to rely on either the software auto-updating itself (through various advanced mechanisms), or in features that never quite made the cut to mainline K8s. 

The command here is relatively simple: get the `deployer` and delete it. It could be reduced further to a single line. However, I'm erring on the side of caution and _only_ attempting to when the deployer is _completed_, which requires additional checks. The issue is that we don't want it deleting a `deployer` when it's still deploying.

Overall there are quite a few things with the GCP Marketplace Tools in general which have room for improvement, but are largely out of our hands. Reducing this "delete deployer" check to a single command was about as simple as I could make it.

Reply:
I appreciate your hard work on this topic! Thank you for taking such a detailed look. I didn't come up with anything better either.

Reply:
I went ahead and moved the "script" to an actual `Config` which makes it a lot more readable. We use a similar mechanism in the verifier scripts already (in `apptest`), so I largely took from that example.